src/main/kotlin/org/rust/ide/intentions/createFromUsage/utils.kt

```diff
@@ -67,7 +67,8 @@ fun getTargetItemForFunctionCall(path: RsPath): CallableInsertionTarget? {
 
 fun insertStruct(targetModule: RsMod, struct: RsStructItem, sourceFunction: RsElement): RsStructItem {
     if (targetModule == sourceFunction.containingMod) {
-        return sourceFunction.parent.addBefore(struct, sourceFunction) as RsStructItem
+        val anchor = sourceFunction.ancestors.firstOrNull { it in targetModule.children } ?: sourceFunction
```

Comment:
```suggestion
        val anchor = sourceFunction.ancestors.firstOrNull { it.parent == targetModule } ?: sourceFunction
```
?

Reply:
:man_facepalming: Yes, that's what I was looking for :D Thanks.